File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/Java_Gradle_Project/Laboratory_1/FestivalTicketSale/src/main/java/org/example/Model/Validators/ValidatorFactory.java
package org.example.Model.Validators;

import org.example.Model.Exceptions.ValidatorException;
import org.example.Model.Validators.Enums.ValidatorStrategy;
import org.example.Model.Validators.Interfaces.Factory;
import org.example.Model.Validators.Interfaces.Validator;
import org.example.Model.Validators.ModelValidators.AngajatValidator;
import org.example.Model.Validators.ModelValidators.SpectacolValidator;

public class ValidatorFactory implements Factory {
    private static ValidatorFactory instance;
    @Override
    public Validator createValidator(ValidatorStrategy validatorStrategy) {
        switch (validatorStrategy)
        {
            case Angajat -> {return new AngajatValidator();}
            case Spectacol -> {return new SpectacolValidator();}
            default -> throw new ValidatorException("Invalid strategy for creating the validator!");
            }

    }
    public static ValidatorFactory getInstance()
    {
        if(instance==null)
        {
            instance=new ValidatorFactory();
        }
        return instance;
    }
}

File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/C#_Project/Laboratory_2/FestivalTicketSales/Model/Validators/ValidatorFactory.cs
/*using System;
using FestivalTicketSale.Model.Exceptions;
using FestivalTicketSale.Model.Validators.Enums;
using FestivalTicketSale.Model.Validators.Interfaces;
using FestivalTicketSale.Model.Validators.ModelValidators;

namespace FestivalTicketSale.Model.Validators
{
    public class ValidatorFactory : IFactory<TE>
    {
        private static ValidatorFactory instance;

        public static ValidatorFactory GetInstance()
        {/*#1#
            if (instance == null)
            {
                instance = new ValidatorFactory();
            }
            return instance;
        }

        public IValidator CreateValidator(ValidatorStrategy validatorStrategy)
        {
            switch (validatorStrategy)
            {
                case ValidatorStrategy.Angajat:
                    return new AngajatValidator();
                case ValidatorStrategy.Spectacol:
                    return new SpectacolValidator();
                default:
                    throw new ValidatorException("Invalid strategy for creating the validator!");
            }
        }
    }
}*/